### PR TITLE
Feat(Firebase): connect firebase to code and add tests

### PR DESCRIPTION
**Pull request by davidholzwarth**
# Summary

- Connected firebase to the project
- Added a test to check if writing to, getting from and deleting from Firebase works.
- Important:  You need to add the google-services.json file that I shared with you on Telegram to the app/ folder locally.
# Details

### CODE REVIEW

File: app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    // Add a test file to the database
+    db.collection("testCollection")
+        .add(testData)
+        .addOnSuccessListener { documentReference ->
+          assertTrue(true)
+          Log.d("FireBaseTest", "DocumentSnapshot added with ID: " + documentReference.id)
+          latchAdd.countDown()
+        }
+        .addOnFailureListener { e ->
+          assertTrue("Write to Firebase failed. Got error: $e", false)
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt:46]: Nitpick: I think fail() method from Junit would be a little cleaner than using a assertTrue(false)

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -18,7 +18,7 @@
             android:label="@string/title_activity_second"
             android:theme="@style/Theme.SampleApp" />
         <activity
-            android:name=".MainActivity"
+            android:name="ch.hikemate.app.MainActivity"
```
Comment (Wylwi) [inline app/src/main/AndroidManifest.xml:21]: Question: Wasn't it working without it?

File: app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    // Iterate through each document in the result
+    for (document in snapshot.documents) {
+      // Check if the document contains the specific key and value
+      if (document.contains(key) && document.getString(key) == value) {
+        return true // Value found
+      }
+    }
+    return false // Value not found
+  }
+}
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt:104]: Haven't found any way to do it better, great job!

File: app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+              .addOnSuccessListener {
+                assertTrue("Delete from Firebase worked. Deleted document: " + document.id, true)
+                Log.d("FireBaseTest", "DocumentSnapshot successfully deleted!")
+              }
+              .addOnFailureListener { e ->
+                assertTrue("Delete from Firebase failed. Got error: $e", false)
+              }
+        }
+      } else {
+        assertTrue("Read from Firebase failed. Got error: " + "${task.exception}", false)
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt:80]: Nitpick: I think fail() method from Junit would be a little cleaner than using a assertTrue(false)

File: app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        for (document in task.result!!) {
+          db.collection("testCollection")
+              .document(document.id)
+              .delete()
+              .addOnSuccessListener {
+                assertTrue("Delete from Firebase worked. Deleted document: " + document.id, true)
+                Log.d("FireBaseTest", "DocumentSnapshot successfully deleted!")
+              }
+              .addOnFailureListener { e ->
+                assertTrue("Delete from Firebase failed. Got error: $e", false)
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt:76]: Nitpick: I think fail() method from Junit would be a little cleaner than using a assertTrue(false)

File: app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+    assertTrue(latchAdd.await(5, TimeUnit.SECONDS))
+
+    // Read the test file from the database
+    db.collection("testCollection").get().addOnCompleteListener { task ->
+      if (task.isSuccessful) {
+        assertTrue(containsAddedValue(task.result, "testKey", testValue))
+      } else {
+        assertTrue("Read from Firebase failed. Got error: " + "${task.exception}", false)
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/FireBaseTest.kt:57]: Nitpick: I think fail() method from Junit would be a little cleaner than using a assertTrue(false)

**Review by Wylwi — APPROVED** [review]
Looks pretty good to me! Really nitpicking on some details, but otherwise great job!

**Review by rBergCode — APPROVED** [review]
LGTM. Just a tip: instead of "chore" you could use "fix" for the nature of your commit ;)

**Pull request merged**